src/rgw/motr/gc/gc.cc

```diff
@@ -223,27 +244,97 @@ uint32_t MotrGC::get_max_indices() {
   return gc_max_indices;
 }
 
-int MotrGC::delete_motr_obj_from_gc(motr_gc_obj_info ginfo) {
+int MotrGC::delete_obj_from_gc(motr_gc_obj_info ginfo) {
+  int rc = delete_motr_obj(ginfo.mobj);
+  return rc;
+}
+
+int MotrGC::process_parts(motr_gc_obj_info ginfo, std::time_t end_time) {
+  int rc = 0;
+  int max_entries = 10000;
+  int number_of_parts = 0;
+  int processed_parts = 0;
+  std::vector<std::string> keys(max_entries);
+  std::vector<bufferlist> vals(max_entries);
+
+  rc = store->next_query_by_name(ginfo.multipart_iname, keys, vals);
+  if (rc < 0) {
+    ldout(cct, 0) <<__func__<<": ERROR: next query failed. rc="
+      << rc << dendl;
+    return rc;  
+  }
+  number_of_parts = rc;
+  for (const auto& bl: vals) {
+    if (bl.length() == 0)
+      break;
+
+    RGWUploadPartInfo info;
+    auto iter = bl.cbegin();
+    info.decode(iter);
+    rgw::sal::Attrs attrs_dummy;
+    decode(attrs_dummy, iter);
+    Meta mobj;
+    mobj.decode(iter);
+    ldout(cct, 20) <<__func__<< ": part_num=" << info.num
+                             << " part_size=" << info.size << dendl;
+    std::string tag = PRIx64 + std::to_string(mobj.oid.u_hi) + ":" +
+                      PRIx64 + std::to_string(mobj.oid.u_lo);
+    std::string part_name = "part.";
+    char buff[32];
+    snprintf(buff, sizeof(buff), "%08d", (int)info.num);
+    part_name.append(buff);
+
+    std::string obj_fqdn = ginfo.name + "." + part_name;
+    motr_gc_obj_info gc_obj(tag, obj_fqdn, mobj, ginfo.deletion_time,
+                            info.size, info.size_rounded, false, "");
+    rc = enqueue(gc_obj);
+    if (rc < 0) {
+      ldout(cct, 0) <<__func__<< ": ERROR: failed to push " 
+                          << obj_fqdn << "into GC queue " << dendl;
+      continue;
+    }
+    processed_parts++;
+    bufferlist bl_del;
+    rc = store->do_idx_op_by_name(ginfo.multipart_iname,
+                                  M0_IC_DEL, part_name, bl_del);
+    if (rc < 0) {
+      ldout(cct, 0) <<__func__<< ": ERROR: failed to remove part " << part_name 
+                      << " from part index " << ginfo.multipart_iname << dendl;
+    }
+    if (std::time(nullptr) > end_time) {
```

Comment:
also need to exit if motr_gc is going down

Reply:
Added